### ISSUE TITLE
Cast or assignment to `Any` not allowed with partially unknown functions

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/pyright#6096]
**Describe the bug**

When I use a `cast(Any, ...)` around a value, or possibly if I assign a value to something local explicitly as `Any`, I expect that I'll be able to avoid the "...is partially unknown" error message under stricter checking settings.

**Code or Screenshots**

```python
# blah.pyi
def yadda(x, y, z) -> None:
    ...

# foo.py
from typing import Any, cast
import blah

blah.yadda(1, 2, 3)

cast(Any, blah.yadda)(1, 2, 3)

yadda: Any = blah.yadda
yadda(1, 2, 3)
```

I'd expect only the first to error, but instead for all 3 I get:

```
Type of "yadda" is partially unknown
  Type of "yadda" is "(x: Unknown, y: Unknown, z: Unknown) -> None"
```

**VS Code extension or command-line**

I'm running on VS Code via Pylance 2023.10.10

**Comment by erictraut** [comment]
I think it's maybe reasonable to suppress this diagnostic with a `cast` call. I'll need to give that more thought. 

I don't think it should be suppressed for the RHS of an assignment. The declared type of the LHS of an assignment shouldn't have that effect on the RHS evaluation.

**Comment by DanielRosenwasser** [comment]
I think the cast case definitely makes sense. For assignments, it's more of a nice to have, since practically speaking `Any` is all I would care about.

**Comment by erictraut** [comment]
Thinking more about this, I don't think that the entire expression in the RHS of the `cast` call should be exempt from "reportUnknownXXX" checks. This is a strict-mode check that is not enabled by default. If you want to prevent this error, you should apply the cast to the first part of the expression that is unknown.

```python
cast(Any, blah).yadda(1, 2, 3)
```

Alternatively, you can use a `# type: ignore` or `# pyright: ignore` comment to suppress the error.

**Comment by DanielRosenwasser** [comment]
Isn't that way less safe? I now have no assurance that `yadda` exists, whereas before I had at least that.

**Comment by erictraut** [comment]
You're generally going to be unsuccessful using strict-mode type checks if you're importing from libraries that lack type information. You can create type stubs to fill in the missing types, and that's going to be way more effective (and type safe) than attempting to insert casts.
